### ISSUE TITLE
EmberJS conflicts with Mautic form JS

### ISSUE THREAD

**Issue by pjcarly** [mautic/mautic#3614]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | ✓
| Feature request? | 
| Enhancement? | 

## Description:
I managed to install Mautic version 2.7, and came to the part of embedding the Javascript in my website, the embedding works without a problem. But i'm getting a JS error, and can't find anything about it online.

```
Uncaught TypeError: MauticJS.postEventDeliveryQueue[i] is not a function
    at HTMLDocument.<anonymous> (mtc.js:14)
    at Object.MauticJS.dispatchEvent (mtc.js:10)
    at mtc.js:70
    at XMLHttpRequest.xhr.onreadystatechange (mtc.js:7)
```
When I look in my JS dev Console, the MauticJS object exists, there is an array at postEventDeliveryQueue, but the array contains 1 element, which is undefined (so I'm guessing there's something wrong there).

I have mautic currently installed on my localhost to test it out, and also running my website from another folder on my localhost

Website: http://localhost/myawesomewebsite/web
Mautic: http://localhost/mautic

For extra information let me know!
## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.7 & 2.7.1
| PHP version | 7.0.13

### Steps to reproduce:
1. Install blank Mautic 2.7
2. Add JS tracking info to website
3. On website load, above error happens
 
### Log errors: 

_Please check for related errors in the latest log file in [mautic root]/app/log/ and/or the web server's logs and post them here. Be sure to remove sensitive information if applicable._ 

No error log whatsoever, probably due to being a JS error

**Comment by escopecz** [comment]
Could you expand the second point with what JS have you used?

**Comment by pjcarly** [comment]
What JS I have used? I'm just embedding the JS according to the documentation (https://mautic.org/docs/en/contacts/contact_monitoring.html) on the website like so:

```
(function(w,d,t,u,n,a,m){w['MauticTrackingObject']=n;
    w[n]=w[n]||function(){(w[n].q=w[n].q||[]).push(arguments)},a=d.createElement(t),
    m=d.getElementsByTagName(t)[0];a.async=1;a.src=u;m.parentNode.insertBefore(a,m)
})(window,document,'script','http://localhost/mautic/mtc.js','mt');

mt('send', 'pageview');
```

**Comment by pjcarly** [comment]
I can confirm the same error when installed on a VPS.

**Comment by pjcarly** [comment]
Still present in 2.7.1

**Comment by mbernabeu** [comment]
I have the same problem

In my case I think the error is by the prototype library
I have tried in many magentos and have the same error since version 2.6.0
I have tried removing the prototype dependency and the error is no longer displayed
But I need to find some solution because prototype is necessary for all versions of magento 1

@pjcarly is a magento your website?

**Comment by pjcarly** [comment]
Hi @mbernabeu no, it's a Drupal 8 website

**Comment by pjcarly** [comment]
I think you're on the right track.

It must have something to do with the libraries being included by our CMS, (probably the prototype you talk about). I just tried including this javascript on a static website (no Drupal) and the event fires just fine.

**Comment by pjcarly** [comment]
An Update. This is not due to Drupal 8, I have narrowed it down to a conflict with EmberJS. I have embedded an EmberJS application on a few pages of my D8 site, and only on those pages Mautic throws this exception.

My guess is that Ember does some prototype overriding. I will try refactoring and building my Ember app without [prototype extensions](https://guides.emberjs.com/v2.12.0/configuring-ember/disabling-prototype-extensions/)
